liblangutil/ParserBase.h

```diff
@@ -62,10 +69,17 @@ class ParserBase
 
 	///@{
 	///@name Helper functions
-	/// If current token value is not _value, throw exception otherwise advance token.
+	/// If current token value is not @a _value, throw exception otherwise advance token
+	//  @a if _advance is true and error recovery is in effect.
 	void expectToken(Token _value, bool _advance = true);
+
+	/// Like expectToken but if there is an error ignores tokens until
+	/// the expected token or EOS is seen. If EOS is encountered, back up to the error point,
+	/// and throw an exception so that a higher grammar rule has an opportunity to recover.
+	void expectTokenOrConsumeUntil(Token _value, std::string const& _currentNodeName, bool _advance = true);
```

Comment:
This is not part of this PR and lacks an implementation in this PR, please remove from here!

Reply:
There is a confusion here part of which is due to the back and for commit and revert that was done. 

This doesn't make sense except in the context of error recovery. That was committed into develop and then pulled. In between I added this commit. When error recovery was reverted, then that broke this. So think of this as errorecovery the full version as we requested as part of the error recovery PR.  

Given where we are now, I don't think it makes sense to have two PRs for error recovery: the first part that was in and reverted and the finishing part to include an field for StandardCompiler.